integration-tests/opentelemetry/src/test/java/org/apache/camel/quarkus/component/opentelemetry/it/OpenTelemetryTest.java

```diff
@@ -56,6 +58,7 @@ public void testTraceRoute() {
         }
 
         // Retrieve recorded spans
+        await().atMost(30, TimeUnit.SECONDS).pollDelay(50, TimeUnit.MILLISECONDS).until(() -> getSpans().size() == 5);
         List<Map<String, String>> spans = getSpans();
```

Comment:
nitpcik if time: expectations might be redundant with await clause ?

Reply:
Yes, I did consider that. But left them in to be extra paranoid in case something miraculously changed in between the await and re-fetching the spans.